server/src/main/java/org/candlepin/async/tasks/HypervisorHeartbeatUpdateJob.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.async.tasks;
+
+import org.candlepin.async.AsyncJob;
+import org.candlepin.async.ArgumentConversionException;
+import org.candlepin.async.JobArguments;
+import org.candlepin.async.JobConfig;
+import org.candlepin.async.JobConfigValidationException;
+import org.candlepin.async.JobExecutionContext;
+import org.candlepin.async.JobExecutionException;
+import org.candlepin.common.filter.LoggingFilter;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.Owner;
+
+import com.google.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Date;
+import java.util.Objects;
+
+
+
+/**
+ * Asynchronous job for refreshing the last check-in time for a specific reporter ID
+ */
+public class HypervisorHeartbeatUpdateJob implements AsyncJob {
+    private static final Logger log = LoggerFactory.getLogger(HypervisorHeartbeatUpdateJob.class);
+
+    public static final String JOB_KEY = "HYPERVISOR_HEARTBEAT_UPDATE";
+    public static final String JOB_NAME = "hypervisor_heartbeat_update";
+
+    private static final String REPORTER_ID = "reporter_id";
+    private static final String OWNER_KEY = "owner_key";
```

Comment:
Maybe this should be `org` instead of `owner_key`, in order to be consistent with the other jobs?
It might be better if we factor out into public flags all the common flags used by jobs like `OWNER_KEY` and `CONSUMER_ID`? It might help in a potential future case where we're using debug SQL with job arguments being standardized instead of some jobs using 'org' and others 'owner_key' as a key.

Reply:
We could use org, but I feel that's more inheritance of the old pinsetter stuff, where it was called "org" specifically because of the parameter name from logback.xml (or the other way around). In this particular case, I want the variable to be clear that it's the org key and not ID or anything else. This is why the UndoImportsJob task includes the LoggingFilter.OWNER_KEY field (which has a value of "org" due to the afore mentioned config requirement) expressly for metadata concerns; we're no longer required to couple the name of our owner variable to what is needed for logging. Since I expect this PR to go in second, I didn't fully expand that variable out like I did in the other PR.

For jobs in general, I want the jobs to be free to use the most expressive argument names deemed necessary by the author -- especially since with the addition of per-job config objects, a lot/all of it is fully encapsulated anyway.